sunpy/net/hek/tests/test_hek.py

```diff
@@ -223,10 +203,156 @@ def test_query_multiple_operators():
+
+
+@pytest.fixture(scope="session")
+def coronal_hole_search_result():
+    client = hek.HEKClient()
+    result = client.search(attrs.Time('2011/08/09 07:23:56', '2011/08/09 12:40:29'),
+                           attrs.hek.EventType('CH'))
+    return result
+
+
+@pytest.mark.remote_data
+def test_compare_event_coords(coronal_hole_search_result):
+    event_coord = SkyCoord(-2.91584*u.arcsec,
+                           940.667*u.arcsec,
+                           frame=Helioprojective(observer=get_earth('2011-08-09 06:00:08.000')))
+    assert coronal_hole_search_result['event_coord'][0] == event_coord
+
+
+@pytest.mark.remote_data
+def test_obs_meanwavel(coronal_hole_search_result):
+    assert coronal_hole_search_result['obs_meanwavel'][0] == 193.0*u.angstrom
+
+
+@pytest.fixture
+def flare_search():
+    return attrs.Time('2011/08/09 07:23:56', '2011/08/09 12:40:29'), attrs.hek.EventType('FL')
+
+
+@pytest.mark.remote_data
+def test_ssw_latest_events_flares(flare_search):
+    result = Fido.search(*flare_search, attrs.hek.FRM.Name == 'SSW Latest Events')
+    assert len(result[0]) == 2
+
+
+@pytest.mark.remote_data
+def test_not_ssw_latest_events_flares(flare_search):
+    result = Fido.search(*flare_search, attrs.hek.FRM.Name != 'SSW Latest Events')
+    assert len(result[0]) == 19
+
+
+@pytest.mark.remote_data
+def test_flares_peak_flux(flare_search):
+    result = Fido.search(*flare_search, attrs.hek.FL.PeakFlux > 4000.0)
+    assert len(result[0]) == 1
+    assert result[0]['fl_peakflux'] > 4000.0*u.DN/(u.pix*u.s)
+
+
+@pytest.mark.remote_data
+def test_flares_peak_flux_and_position(flare_search):
+    result = Fido.search(*flare_search,
+                         attrs.hek.Event.Coord1 > 800,
+                         attrs.hek.FL.PeakFlux > 1000)
+    assert len(result[0]) == 7
+    assert all([c.Tx > 800*u.arcsec for c in result[0]['event_coord']])
+    assert (result[0]['fl_peakflux'] > 1000.0*u.DN/(u.pix*u.s)).all()
+
+
+@pytest.mark.remote_data
+def test_flares_python_logical_ops(flare_search):
+    result = Fido.search(*flare_search,
+                         (attrs.hek.Event.Coord1 > 50) and (attrs.hek.FL.PeakFlux > 1000))
+    assert len(result[0]) == 7
+    assert all([c.Tx > 50*u.arcsec for c in result[0]['event_coord']])
+    assert (result[0]['fl_peakflux'] > 1000.0*u.DN/(u.pix*u.s)).all()
+
+
+@pytest.mark.remote_data
+@pytest.mark.parametrize("event_type", [
+    'AR',
+    'CE',
+    'CD',
+    'CW',
+    'FI',
+    'FE',
+    'FA',
+    'LP',
+    'OS',
+    'SS',
+    'EF',
+    'CJ',
+    'PG',
+    'OT',
+    'NR',
+    'SG',
+    'SP',
+    'CR',
+    'CC',
+    'ER',
+    'TO',
+    'HY',
+    'BU',
+    'EE',
+    'PB',
+    'PT',
+])
+def test_event_types(event_type):
+    # Smoke test for all event types
+    _ = Fido.search(attrs.Time('2017/09/06 11:59:04', '2017/09/06 17:05:04'), attrs.hek.EventType(event_type))
+
+
+@pytest.mark.remote_data
+def test_raw_hek_result_preserved(hek_result):
```

Comment:
codecov reporting unconvered lines on this test makes me thing it's not doing its job properly.